app/services/catalog/create_request_body_from.rb

```diff
@@ -0,0 +1,30 @@
+module Catalog
+  class CreateRequestBodyFrom
+    attr_reader :result
+
+    def initialize(order, order_item, task)
+      @order = order
+      @order_item = order_item
+      @task = task
+    end
+
+    def process
+      svc_params_sanitized = Catalog::OrderItemSanitizedParameters.new(:order_item_id => @order_item.id).process.sanitized_parameters
+      local_tag_resources = Tags::CollectLocalOrderResources.new(:order_id => @order.id).process.tag_resources
+      remote_tag_resources = Tags::CollectRemoteInventoryResources.new(@task).process.tag_resources
+
+      @result = ApprovalApiClient::RequestIn.new.tap do |request|
+        request.name      = @order_item.portfolio_item.name
+        request.content   = {
+          :product   => @order_item.portfolio_item.name,
+          :portfolio => @order_item.portfolio_item.portfolio.name,
+          :order_id  => @order_item.order_id.to_s,
+          :params    => svc_params_sanitized
+        }
+        request.tag_resources = local_tag_resources + remote_tag_resources
```

Comment:
Strictly for easier understanding, I would like to see `local_tag_resources + remote_tag_resources` moved into a method called something like `all_required_tags` or `all_known_tags`. Largely that it's hard to understand that we are sending all remote tag resources ( Topology ) + Catalog's local_tags over to Approval, and that combination of the two is what Approval uses.

I missed the subtlety of this line before, so making it obvious may help others later.